src/phoenix/utilities/security.py

```diff
@@ -0,0 +1,24 @@
+from typing import Mapping, Optional
+
+from starlette.datastructures import URL
+
+from phoenix.config import get_env_csrf_trusted_origins, get_web_base_url
+
+
+def get_referer(headers: Mapping[str, str]) -> Optional[str]:
+    """
+    Extracts the referer URL from the request headers. Returns the referer URL if it is
+    trusted, otherwise returns None.
+    """
+    if not (referer := headers.get("referer")):
+        return None
+    if URL(referer).hostname in ("localhost", "127.0.0.1", "0.0.0.0"):
+        return referer
+    base_url = URL(get_web_base_url())
+    if referer.startswith(f"{base_url.scheme}://{base_url.hostname}"):
+        return referer
+    origins = get_env_csrf_trusted_origins()
```

Comment:
Possible to avoid the pattern of reading env vars inside the function? Would be nice to read the env vars once and keep the functions pure.